src/multipart.lua

```diff
@@ -103,23 +126,28 @@ local function decode(body, boundary)
             local pos = v:match("^%s*[Nn][Aa][Mm][Ee]=()")
             if pos then
               local current_value = v:match("^%s*([^=]*)", pos):gsub("%s*$", "")
-              part_name = string.sub(current_value, 2, string.len(current_value) - 1)
+              part_name = sub(current_value, 2, #current_value - 1)
             end
           end
         end
-        table.insert(part_headers, line)
+
+        insert(part_headers, line)
+
       else
-        if is_header(line) then
+        if not processing_part_value and is_header(line) then
```

Comment:
if `not processing_part_value` is true, then there's not need to set it to false on the following line